python/hail/docs/hailpedia/matrixtable.rst

```diff
@@ -0,0 +1,404 @@
+    Column fields:
+        's': str
+    ----------------------------------------
+    Row fields:
+        'locus': locus<GRCh37>
+        'alleles': array<str>
+        'rsid': str
+        'qual': float64
+        'filters': set<str>
+        'info': struct {
+            NEGATIVE_TRAIN_SITE: bool,
+            AC: array<int32>,
+            ...
+            DS: bool
+        }
+    ----------------------------------------
+    Entry fields:
+        'GT': call
+        'AD': array<int32>
+        'DP': int32
+        'GQ': int32
+        'PL': array<int32>
+    ----------------------------------------
+    Column key:
+        's': str
+    Row key:
+        'locus': locus<GRCh37>
+        'alleles': array<str>
+    Partition key:
+        'locus': locus<GRCh37>
+    ----------------------------------------
+
+Common Operations
+=================
+
+Like tables, Hail provides a number of useful methods for manipulating data in a
+matrix table.
+
+**Filter**
+
+:class:`.MatrixTable` has three methods to filter based on expressions:
+
+- :meth:`.MatrixTable.filter_rows`
+- :meth:`.MatrixTable.filter_cols`
+- :meth:`.MatrixTable.filter_entries`
+
+Filter methods take a :class:`.BooleanExpression` argument. These expressions
+are generated by applying computations to the fields of the matrix table:
+
+    >>> filt_mt = mt.filter_rows(hl.len(mt.alleles) == 2)
+
+    >>> filt_mt = mt.filter_cols(hl.agg.mean(mt.GQ) < 20)
+
+    >>> filt_mt = mt.filter_entries(mt.DP < 5)
+
+These expressions can compute arbitrarily over the data: the :meth:`.MatrixTable.filter_cols`
+example above aggregates entries per column of the matrix table to compute the
+mean of the `GQ` field, and removes columns where the result is smaller than 20.
+
+**Annotate**
+
+:class:`.MatrixTable` has four methods to add new fields or update existing fields:
+
+- :meth:`.MatrixTable.annotate_rows`
+- :meth:`.MatrixTable.annotate_cols`
+- :meth:`.MatrixTable.annotate_entries`
+- :meth:`.MatrixTable.annotate_globals`
+
+Annotate methods take keyword arguments where the key is the name of the new
+field to add and the value is an expression specifying what should be added.
+
+The simplest example is adding a new global field `foo` that just contains the constant
+5.
+
+    >>> mt_new = mt.annotate_globals(foo = 5)
+    >>> print(mt.globals.dtype.pretty())
+    struct {
+        foo: int32
+    }
+
+Another example is adding a new row field `call_rate` which computes the fraction
+of non-missing entries `GT` per row:
+
+    >>> mt_new = mt.annotate_rows(call_rate = hl.agg.fraction(hl.is_defined(mt.GT)))
+
+Annotate methods are also useful for updating values. For example, to update the
+GT entry field to be missing if `GQ` is less than 20, we can do the following:
+
+    >>> mt_new = mt.annotate_entries(GT = hl.case()
+    ...                                     .when(mt.GQ >= 20, mt.GT)
+    ...                                     .or_missing())
```

Comment:
we should probably take this chance to encourage `hl.or_missing(mt.GQ >= 20, mt.GT)`